src/main/java/gregtech/api/capability/impl/MultiblockRecipeLogic.java

```diff
@@ -103,6 +103,25 @@ protected IMultipleTankHandler getOutputTank() {
         return controller.getOutputFluidInventory();
     }
 
+    @Override
+    protected boolean canWorkWithInputs() {
+        MultiblockWithDisplayBase controller = (MultiblockWithDisplayBase) metaTileEntity;
+        if (controller instanceof RecipeMapMultiblockController) {
+            RecipeMapMultiblockController distinctController = (RecipeMapMultiblockController) controller;
+
+            if (distinctController.canBeDistinct() && distinctController.isDistinct()) {
+                if (invalidatedInputList.isEmpty() || !metaTileEntity.getNotifiedFluidInputList().isEmpty()) return true;
+                for (IItemHandlerModifiable bus : metaTileEntity.getNotifiedItemInputList()) {
+                    if (invalidatedInputList.contains(bus)) {
+                        return true;
```

Comment:
As well as returning true, should we remove the selected bus from the `invalidatedInputList`?

In fact, we might want to delay returning true until we finish looping over all the buses, so that we can remove any that were notified from `invalidatedInputList` and then only return true if there were buses that were removed from the list.